src/state/slices/opportunitiesSlice/thunks.ts

```diff
@@ -1,14 +1,15 @@
 import type { StartQueryActionCreatorOptions } from '@reduxjs/toolkit/dist/query/core/buildInitiate'
 import type { AccountId, ChainId } from '@shapeshiftoss/caip'
 import { fromAccountId } from '@shapeshiftoss/caip'
-import { store } from 'state/store'
+import type { AppDispatch } from 'state/store'
 
 import { foxEthStakingIds } from '../opportunitiesSlice/constants'
 import { CHAIN_ID_TO_SUPPORTED_DEFI_OPPORTUNITIES } from './mappings'
 import { opportunitiesApi } from './opportunitiesApiSlice'
 import { DefiProvider, DefiType } from './types'
 
 export const fetchAllLpOpportunitiesMetadataByChainId = async (
+  dispatch: AppDispatch,
```

Comment:
Was this passed around to avoid circular deps? `store.dispatch` effectively is the same

Reply:
correct yeah, we cannot consume `import { store } from 'state/store'` from these due to circ deps.